### ISSUE TITLE
`isinstance(X, float)` wrongly consumes `bool` type

### ISSUE THREAD

**Issue by tusharsadhwani** [python/mypy#12114]
In the following example, adding an unrelated `isinstance` check causes the code to raise an error.

**To Reproduce**

Code:

```python
from typing import Union

def get_type_name(value: Union[bool, str]) -> str:
    if value is True:
        return "Boolean"
    if value is False:
        return "Boolean"
    return "String"
```

Mypy output:

```console
$ mypy --strict testcode.py
Success: no issues found in 1 source file
```

Adding an `isinstance(value, float)` check:

```python
from typing import Union

def get_type_name(value: Union[bool, str]) -> str:
    if isinstance(value, float):  # a completely unrelated check
        return "Number"

    if value is True:
        return "Boolean"
    if value is False:
        return "Boolean"

    return "String"
```

Mypy output:

```console
$ mypy --strict testcode.py
testcode.py:5: error: Missing return statement
testcode.py:9: error: Non-overlapping identity check (left operand type: "str",
right operand type: "Literal[True]")
testcode.py:11: error: Non-overlapping identity check (left operand type: "str",
 right operand type: "Literal[False]")
Found 3 errors in 1 file (checked 1 source file)
```

Mypy says `value is True` and `value is False` are non overlapping checks, but it works:

```pycon
$ python -i testcode.py
>>> get_type_name(5.0)
'Number'
```

**Your Environment**

- Mypy version used: 0.931
- Mypy command-line flags: `--strict`
- Python version used: Python 3.9 and 3.10
- Operating system and version: Linux, Pop!_OS 21.10

**Comment by JelleZijlstra** [comment]
This is probably because bool is a subclass of int (in reality) and int is a subclass of float (in the type system).

**Comment by hauntsaninja** [comment]
mypy could maybe have an option to pretend that bool isn't a subclass of int, in which case this is a duplicate of https://github.com/python/mypy/issues/8363
